nav2_mission_executor/README.md

```diff
@@ -0,0 +1,82 @@
+# Mission Executor
+
+The Mission Executor module is a task server that coordinates other tasks. It can be used to direct the activities multiple robots as well as other, non-robot tasks.
+
+## Overview
+ 
+The Mission Executor task server receives a **MissionPlan** and returns whether the plan was successfully executed, failed, or was canceled.
+
+```C++
+namespace nav2_tasks {
+
+using ExecuteMissionCommand = nav2_msgs::msg::MissionPlan;
+using ExecuteMissionResult = std_msgs::msg::Empty;
+
+using ExecuteMissionTaskClient = TaskClient<ExecuteMissionCommand, ExecuteMissionResult>;
+using ExecuteMissionTaskServer = TaskServer<ExecuteMissionCommand, ExecuteMissionResult>;
+
+} //  namespace nav2_tasks
+```
+
+NOTE: An Empty message is used in the definition of the task client and server if there is no other data required.
+
+The [mission plan message](https://github.com/ros-planning/navigation2/blob/master/nav2_msgs/msg/MissionPlan.msg) consists of a header and a mission plan string:
+
+```
+std_msgs/Header header
+string mission_plan
```

Comment:
This line with just `string mission_plan` scares me quite a bit. This package seems to assume that your intended waypoints are in this xml format (or is in a file at all) and then serializes into this string. If I'm a user and I'm updating waypoints, it should be as easy as updating the MissionPlan message with my new intended waypoints. I can't think of many situations where I have just a list of waypoints I want to visit in order.  

The process right now would be: new points -> convert into some xml -> serialize -> load into message.
As  the process would be that makes most sense to me: new points -> populate vector of messages -> load into message.

I strongly believe this `mission_plan` needs to be an actual array of waypoints by a mission plan msg definition. This string can't be easily tested or intercepted and debugged. More or less wouldn't work with CLI tools.

Reply:
I concur -- in ROS1, the use of a giant string was usually the sign of a design that hadn't been fully thought out. It makes the API easy to misuse, and often hard to use correctly.

Reply:
@mjeronimo -- I have no issue with the usage of the behavior tree -- my only concern is the XML. I'd rather see something that is composed of actual messages with compile-time known types. I realize the recursive nature of the behavior trees might get complex, but it should be possible to create it as a fixed message type and avoid the XML altogether (as well as giving lots of compile time checking and an object oriented approach to building the mission plan).

Reply:
I agree with this comment @mikeferguson. these mesage types are somewhat "required" in my head due to the nature of applications that aren't the direct output of a behavior tree that were dispatched by other clients (human CLI, other AI based optimizers).

Reply:
@SteveMacenski Thanks for opening the ticket. Let's move the discussion over there as well as, perhaps, the weekly meeting. For now, I'll integrate this PR as it describes what we're currently doing and we can discuss further.

Reply:
Sounds good